lib/rubocop/cop/lint/ambiguous_block_association.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for ambiguous block association with method
+      # when param passed without parentheses.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # It's ambiguous because there is no parentheses around `a` param
+      #   some_method a { |val| puts val }
+      #
+      # @example
+      #
+      #   # good
+      #
+      #   # With parentheses, there's no ambiguity.
+      #   some_method(a) { |val| puts val }
+      class AmbiguousBlockAssociation < Cop
+        MSG = 'Parenthesize the param `%s` to make sure that block will be '\
+              'associated with `%s` method call.'.freeze
+
+        def on_send(node)
+          _klass, method_name, args = node.children
```

Comment:
There's a `SendNode` extension, so you no longer need to destructure send nodes. I'll highlight below how to use it. 🙂